src/app-layout/drawer/index.tsx

```diff
@@ -140,7 +140,7 @@ interface DrawerTriggerProps {
   badge: boolean | undefined;
   itemId?: string;
   isActive: boolean;
-  trigger: PublicDrawer['trigger'];
+  trigger: AppLayoutProps.Drawer['trigger'];
```

Comment:
Following up on this one: https://github.com/cloudscape-design/components/pull/1697#discussion_r1374751945